python/tchannel/tornado/peer.py

```diff
@@ -551,22 +548,40 @@ def send(self, arg1, arg2, arg3,
         if request.is_streaming_request:
             request.ttl = 0
 
+        response = yield self.send_with_retry(
+            request, peer, attempt_times, retry_delay)
+
+        log.debug("Got response %s", response)
+        raise gen.Return(response)
+
+    @gen.coroutine
+    def _send(self, connection, req):
+        response_future = connection.send_request(req)
+        with timeout(response_future, req.ttl):
+            response = yield response_future
+
+        raise gen.Return(response)
```

Comment:
No need for this whole thing to be a coroutine. Just do:

    return gen.with_timeout(connection.send_request(req), req.ttl)